src/metabase/lib/options.cljc

```diff
@@ -89,4 +89,4 @@
   [mbql-clause-or-map]
   (update-options mbql-clause-or-map (fn [options]
                                        (cond-> options
-                                         (not (:lib/uuid options)) (assoc :lib/uuid (str (random-uuid)))))))
+                                         (not (:lib/uuid options)) (assoc :lib/uuid (random-uuid))))))
```

Comment:
Generating and checking real UUIDs is much simpler, I wonder if need these to be strings because of serialization or whatever.